src/impl_views/conversions.rs

```diff
@@ -221,6 +253,22 @@ where
         }
     }
 
+    /// Return the array’s data as a slice, if it is contiguous.
+    /// Otherwise return self in the Err branch of the result.
+    fn try_into_slice_memory_order(self) -> Result<&'a mut [A], Self> {
```

Comment:
I guess it's even harder to make consistent, but I think this method is unfortunately generally useful. I've contemplated adding the corresponding one on ArrayBase (where the Err case would be `&mut Self`). So we should consider if this isn't the best one (or an additional one) to expose for ArrayViewMut.

Reply:
What if we change all the `.as_slice*`/`.to_slice*`/`.into_slice*` methods return a `Result`? Even if the `Err` variant isn't especially useful for all the methods, it'll just be a reference (i.e. pointer size) for all but the `.into_slice*` methods, so it won't cost much.